tfhe/src/high_level_api/keys/inner.rs

```diff
@@ -447,3 +450,50 @@ impl IntegerCompressedCompactPublicKey {
         }
     }
 }
+
+pub struct IntegerServerKeyConformanceParams {
+    pub sk: PBSParameters,
+    pub cpk_param: Option<KeySwitchingKeyConformanceParams2>,
+    pub compression_param: Option<CompressionParameters2>,
+}
+
+impl ParameterSetConformant for IntegerServerKey {
+    type ParameterSet = IntegerServerKeyConformanceParams;
+
+    fn is_conformant(&self, parameter_set: &Self::ParameterSet) -> bool {
+        let Self {
+            key,
+            cpk_key_switching_key_material,
+            compression_key,
+            decompression_key,
+        } = self;
+
+        let cpk_key_switching_key_material_is_ok = match (
+            parameter_set.cpk_param.as_ref(),
+            cpk_key_switching_key_material.as_ref(),
+        ) {
+            (None, None) => true,
+            (Some(cpk_param), Some(cpk_key_switching_key_material)) => {
+                cpk_key_switching_key_material.is_conformant(cpk_param)
+            }
+            _ => false,
```

Comment:
here you could early return from the function since there is no point in doing the other checks if you already know that it should return false

Reply:
Done